WordPress/Classes/Services/PostCoordinator.swift

```diff
@@ -600,14 +603,16 @@ class PostCoordinator: NSObject {
             worker.error = error
             postDidUpdateNotification(for: operation.post)
 
-            if !PostCoordinator.isTerminalError(error) {
-                let delay = worker.nextRetryDelay
-                worker.retryTimer = Timer.scheduledTimer(withTimeInterval: delay, repeats: false) { [weak self, weak worker] _ in
-                    guard let self, let worker else { return }
-                    self.didRetryTimerFire(for: worker)
-                }
-                worker.log("scheduled retry with delay: \(delay)s.")
+            if PostCoordinator.isTerminalError(error) {
+                worker.setLongerDelay()
+            }
+
+            let delay = worker.nextRetryDelay
```

Comment:
I think it's safer to retry (but with a longer delay) than to never retry again.

Reply:
It also helps if you have a draft post where sync is failing due to the media uploads and you go to "Context Menu / Publish / Media Uploads" and remove the upload for there – it doesn't trigger re-sync (now that I wrote it maybe it also should?)